### PR TITLE
specifications: v1 history to v2 releases

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: specifications/history-to-releases.org [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#+END_SRC
+
+On output, the following changes shall be made:
+
+- Acknowledge that a Snap Revision can be compromised of multiple
+  architectures, changing the output of the table from /Arch/ to /Arches/
+  and allowing for a comma separated list.
+- Always displaying the full channel under /Channels/ (the history
+  endpoint return only <risk> for defaults).
+- The /Channels/ comma separated list shall have no spaces for easier
```
Comment (cjp256) [inline specifications/history-to-releases.org:32]: nit: hypenate "comma-separated" or "comma-deliminated".

File: specifications/history-to-releases.org [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+5476    2020-09-11T15:50:57Z  arm64     4.3                       latest/stable*,latest/candidate*,latest/edge
+5475    2020-09-11T13:11:08Z  i386      4.3                       latest/stable*,latest/candidate*,latest/edge
+5474    2020-09-11T13:05:33Z  ppc64el   4.3                       latest/stable*,latest/candidate*,latest/edge
+5473    2020-09-11T12:55:28Z  s390x     4.3                       latest/stable*,latest/candidate*,latest/edge
+5472    2020-09-11T12:50:35Z  amd64     4.3                       latest/stable*,latest/candidate*,latest/edge
+#+END_SRC
+
+* Implementation
+
+The Snap Store API for [[https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-releases][Releases]] returns two /streams/ of interest, those
```
Comment (cjp256) [inline specifications/history-to-releases.org:105]: _M-x fill-pargraph_ :wink:

Comment (cjp256) [inline specifications/history-to-releases.org:105]: s/of interest, those are.../of interest: =revisions= and =releases=.

File: specifications/history-to-releases.org [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+are =revisions= and =releases=. These streams are ordered and relied on
```
Comment (cjp256) [inline specifications/history-to-releases.org:106]: ordered how?  I don't think you need to state that it's relied on (anything in the API must be reliable).

File: specifications/history-to-releases.org [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+5475    2020-09-11T13:11:08Z  i386     4.3                       edge, candidate*, stable*
+5474    2020-09-11T13:05:33Z  ppc64el  4.3                       edge, candidate*, stable*
+5473    2020-09-11T12:55:28Z  s390x    4.3                       edge, candidate*, stable*
+5472    2020-09-11T12:50:35Z  amd64    4.3                       edge, candidate*, stable*
+
+#+END_SRC
+
+And this is how the new output shall display:
+
+#+BEGIN_SRC
```
Comment (cjp256) [inline specifications/history-to-releases.org:72]: The example should probably capture all of the changes (e.g. multi-arch).  And maybe used a more contrived example to keep it succinct?

File: specifications/history-to-releases.org [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+5477    2020-09-11T18:41:18Z  armhf    4.3                       edge, candidate*, stable*
+5476    2020-09-11T15:50:57Z  arm64    4.3                       edge, candidate*, stable*
+5475    2020-09-11T13:11:08Z  i386     4.3                       edge, candidate*, stable*
+5474    2020-09-11T13:05:33Z  ppc64el  4.3                       edge, candidate*, stable*
+5473    2020-09-11T12:55:28Z  s390x    4.3                       edge, candidate*, stable*
+5472    2020-09-11T12:50:35Z  amd64    4.3                       edge, candidate*, stable*
+
+#+END_SRC
+
+And this is how the new output shall display:
```
Comment (cjp256) [inline specifications/history-to-releases.org:70]: nit: "And this is how" is very informal.

File: todo.org [hunk 6]
```diff
@@ -85,3 +85,9 @@ CLOSED: [2020-10-01 jue 10:00]
 - [X] Generate =configure= hook =command-chain= in desktop parts
 - [X] Extend yaml for neon, gnome-2-28 and gnome-3-34 to use desktop
       part's =command-chain= for the =configure= hook
+
+*** TODO [[file:specifications/history-to-releases.org][Snap Store API migration from v1 history to v2 releases]] [0/3]
+
+- [ ] Add bindings to Snap Store package
+- [ ] Replace use of history with releases bindings for CLI
+- [ ] Remove v1 history bindings
```
Comment (cjp256) [inline todo.org:93]: I imagine there'd be a todo item for changing the output format?  or is that coupled with history->release bindings?

Comment (sergiusens) [reply todo.org:93]: I implied it with CLI, second item

File: specifications/history-to-releases.org (hunk 5 above)
Comment (sergiusens) [reply specifications/history-to-releases.org:70]: What is the formal suggestion?

File: specifications/history-to-releases.org (hunk 4 above)
Comment (sergiusens) [reply specifications/history-to-releases.org:72]: I wanted to use a real world example to not make things up and for proper verification, I guess use made up data if that is desired

File: specifications/history-to-releases.org (hunk 2 above)
Comment (sergiusens) [reply specifications/history-to-releases.org:105]: how about using the suggestion :-)

Comment (sergiusens) [reply specifications/history-to-releases.org:105]: I did, the link gets to you :-P

Comment (cjp256) [reply specifications/history-to-releases.org:105]: ahhh

**Review by cjp256 — APPROVED** [review]

**Pull request merged**